app/gui2/src/util/ast/__tests__/aliasAnalysis.test.ts

```diff
@@ -0,0 +1,319 @@
+/**
+ * The testing process employs a unique DSL-like syntax to denote the anticipated results.
+ *
+ * Take this Enso program as an example:
+ * ```
+ * main =
+ *     x = 1
+ *     y = 2
+ *     z = x -> x + y
+ * ```
+ *
+ * It would be annotated in the following manner:
+ * ```
+ * main =
+ *     «1,x» = 1
+ *     «2,y» = 2
+ *     z = «3,x» -> «3,x» + »2,y«
+ *     u = »1,x« + »2,y«
+ * ```
+ *
+ * In general: `»1,x«` indicates that the identifier links to the `x` variable as per the `«1,x»` pattern.
+ * Numerical prefixes are used to differentiate variables of the same name, which is permitted by the language via
+ * shadowing.
+ *
+ * All unannotated identifiers are assumed to preexist in the environment (captured from an external scope or imports).
+ */
+
+import { assert, assertDefined, assertEqual, assertLength, assertNotEqual } from '@/util/assert'
```

Comment:
For tests, you should use assertions provided by vitest. The in-project asserts are meant to be used in application code, and therefore are very minimal.

Reply:
> For tests, you should use assertions provided by vitest. The in-project asserts are meant to be used in application code, and therefore are very minimal.

The issue with the vitest-provided assertions is that they do not have these `asserts` clauses, so the typechecker doesn't really understand them. This forces more code clutter (first asserting that something is defined, and then checking this again for the typechecker) and seems to be a known-issue resulting from the TS language limitations. https://github.com/vitest-dev/vitest/issues/2883